src/ooxml/java/org/apache/poi/openxml4j/opc/ZipPackage.java

```diff
@@ -239,8 +239,8 @@ private static ZipEntrySource openZipEntrySourceStream(ThresholdInputStream zis)
         }
 
         if (this.zipArchive == null) {
-            return this.partList.values().toArray(
```

Comment:
Is partList.sortedValues() an expensive call? If so, the result should be saved in a temp variable so it doesn't have to be called twice.

Does this work?

    return this.partList.sortedValues().toArray(
        new PackagePart[this.partList.size()]);